packages/@react-stately/slider/src/useSliderState.ts

```diff
@@ -159,10 +159,13 @@ export function useSliderState(props: SliderStateOptions): SliderState {
     minValue = DEFAULT_MIN_VALUE,
     maxValue = DEFAULT_MAX_VALUE,
     numberFormatter: formatter,
-    step = DEFAULT_STEP_VALUE,
-    pageSize = Math.max((maxValue - minValue) / 10, step)
+    step = DEFAULT_STEP_VALUE
   } = props;
 
+  // Page step should be at least equal to step and always a multiple of the step.
+  let pageSize = Math.max((maxValue - minValue) / 10, step);
+  pageSize = snapValueToStep(pageSize, pageSize - (pageSize % step), pageSize + step - (pageSize % step), step);
```

Comment:
_think_ this is correct now, previous iteration didn't take into account if step was greater than the max or less than the min. Tossed it into the snapValueToStep in case there are any precision problems